Python/optimizer_analysis.c

```diff
@@ -695,6 +695,11 @@ _Py_uop_analyze_and_optimize(
 )
 {
     OPT_STAT_INC(optimizer_attempts);
+    /* Make sure we have enough C stack space for the optimizer */
+    int margin = 1 + sizeof(JitOptContext)/_PyOS_STACK_MARGIN_BYTES;
+    if (_Py_ReachedRecursionLimitWithMargin(_PyThreadState_GET(), margin)) {
```

Comment:
To work properly, the `optimize_uops` function or this function must be marked `Py_NO_INLINE` as well.

Otherwise, the compiler can just inline the functions and hoist the alloca above the check, making the check useless.